lib/pbench/common/__init__.py

```diff
@@ -29,22 +29,32 @@ def wait_for_uri(uri: str, timeout: int):
                   attempts to connect to the URI
 
     Raises:
-        BadConfig : when the URI does not contain either a host or port
+        BadConfig : when the URI is missing the host or when the scheme is
+                    missing or unrecognized
         ConnectionRefusedError : after the timeout period has been exhausted
     """
     url = urlparse(uri)
     if not url.hostname:
         raise BadConfig("URI must contain a host name")
-    if not url.port:
-        raise BadConfig("URI must contain a port number")
+    if url.scheme == "https":
+        port = 443
+    elif url.scheme == "http":
+        port = 80
+    elif not url.scheme:
+        raise BadConfig("URI must include the scheme ('http://' or 'https://')")
+    else:
+        raise BadConfig(f"URI contains an unrecognized scheme, {url.scheme!r}")
```

Comment:
I don't see any meaningful difference here: whether you skipped the scheme or included a bad one, the `elif` message covers it more usefully than the `else` so they might as well be collapsed.